### PR TITLE
Avoid import-time dependency on jax.experimental

### PR DESCRIPTION
**Pull request by jakevdp**
This was causing circular imports in my static typing explorations.

### CODE REVIEW

File: jax/experimental/array.py [hunk 1]
```diff
@@ -401,3 +402,5 @@ def _array_local_result_handler(aval, sharding, indices):
   return lambda bufs: Array(aval, sharding, bufs, committed=True)
 pxla.local_result_handlers[(core.ShapedArray, pxla.OutputType.Array)] = _array_local_result_handler
 pxla.local_result_handlers[(core.ConcreteArray, pxla.OutputType.Array)] = _array_local_result_handler
+
+ndarray.register(array)
```
Comment (yashk2810) [inline jax/experimental/array.py:406]: `Array` the class? 

nit: Also add a new line at the bottom?

Comment (yashk2810) [reply jax/experimental/array.py:406]: How did the tests not fail?

Comment (yashk2810) [reply jax/experimental/array.py]: Ohh.. the tests did fail!

Comment (jakevdp) [reply jax/experimental/array.py]: Fixed

File: jax/experimental/array.py [hunk 2]
```diff
@@ -401,3 +402,5 @@ def _array_local_result_handler(aval, sharding, indices):
   return lambda bufs: Array(aval, sharding, bufs, committed=True)
 pxla.local_result_handlers[(core.ShapedArray, pxla.OutputType.Array)] = _array_local_result_handler
 pxla.local_result_handlers[(core.ConcreteArray, pxla.OutputType.Array)] = _array_local_result_handler
+
+ndarray.register(Array)
```
Comment (yashk2810) [inline jax/experimental/array.py:406]: Small nit: Can you move this above to line 358 please?

**Review by yashk2810 — APPROVED** [review]

File: jax/experimental/array.py (hunk 2 above)
Comment (jakevdp) [reply jax/experimental/array.py]: Done

**Pull request merged**